addons/website/static/src/js/text_processing.js

```diff
@@ -554,3 +554,13 @@ export function getOffsetNode(textEl, offset) {
     }
     return [offsetNode, offset - index];
 }
+
+/**
+ * It is mainly used to check if an element is part of a text option structure.
+ *
+ * @param {HTMLElement} el
+ * @returns {HTMLElement}
+ */
+export function getTextOptionTarget(el) {
+    return el.closest?.(".o_text_highlight, .o_animated_text");
+}
```

Comment:
This is kinda weird to have this actually: we have a system (`data-selector`) to make it so parts of the DOM are linked to some options. Here this is a new function which is about "finding the parts of the DOM that are linked to text options". Should not we adapt the system so that:
- An *option* can declare itself as a "Text" option
- Adapt our `globalSelector` functions to be able to only consider "Text" options and here it would just be about doing `globalSelector.is($(el), ...only_text...)` 

If not, I also see that we have `data-text-selector` which could be used :shrug:

Reply:
Code updated to use `globalSelector` which seems better, especially if other text options are added. Just let me know if something needs to be changed here.